l2l/utils/trajectory.py

```diff
@@ -16,19 +16,14 @@ class Trajectory:
     def __init__(self, **keyword_args):
         """
         Initializes the trajectory. Some parameters are kept to match the interface with the pypet trajectory.
-        TODO: remove all irrelevant attributes and simplify the class
         """
         if 'name' in keyword_args:
             self._name = keyword_args['name']
         self._timestamp = time.time()
-        self._parameters = ParameterDict(self)  # Contains all parameters
-        self._results = {}  # Contains all results
-        self.individual = Individual()
+        self.parameters = ParameterDict(self)  # Contains all parameters
         self.results = ResultGroup()
-        self.results.f_add_result_group('all_results', "Contains all the results")
         self.current_results = {}
-        self._parameters.parameter_group = {}
-        self._parameters.parameter = {}
+        self.individual = Individual()
         self.individuals = {}
```

Comment:
I think we should make individuals optional. Or more clearer keeping the history of all individuals should be optional. For example later on the specific individual will be added into the individuals dictionary per generation which can make a big overhead if the individual is big.